Topic1/src/com/topic1/ShoppingCart.java

```diff
@@ -0,0 +1,70 @@
+package com.topic1;
+
+import java.util.HashSet;
+//import java.util.HashSet;
+//import java.util.Iterator;
+import java.util.Iterator;
+import java.util.Scanner;
+
+public class ShoppingCart {
+
+	private PayForm pForm;
+	private HashSet<Item> items;
+	private boolean payed;
+	Counter counter = Counter.getInstance();
```

Comment:
[High] you don't need to store the counter instance in a class variable.
*Counter* is a singleton (just one reference). If you store it in a class variable, you will have a reference for each Shopping Cart.